### ISSUE TITLE
Gradient clipping in the wrong place?

### ISSUE THREAD

**Issue by KinWaiCheuk** [jongwook/onsets-and-frames#24]
In your train.py [line 101](https://github.com/jongwook/onsets-and-frames/blob/f5f5bc812a45d88f029452e52ad76ff742626ec3/train.py#L101), you have the gradient clipping after the `.backward()` and `.step()` operations.

Shouldn't we put the `clip_grad_norm_` in between `.backward()` and `.step()`?

**Comment by jongwook** [comment]
Yeah it definitely should. I'm not sure why I did like that. In practice the gradient was well below the clip value during most of the training though.
